### PR TITLE
fix(DIA-198): add better disabled control on login via email screen

### PR DESCRIPTION
**Pull request by araujobarret**
This PR resolves [DIA-198] 

### Description

Related to #9531, prevents submitting the form without being correctly filled via the keyboard submit button.
Also, disable the button until the form is valid.

Bonus: refactored the deprecated use of our test suite within the tests

Before

https://github.com/artsy/eigen/assets/15792853/d01cf52d-676c-4585-88d4-46993f14a92e

After

https://github.com/artsy/eigen/assets/15792853/b2b19bc5-c5cb-456b-9a6b-a20beb43d733

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- fix: disable the cta on the login screen until the form is in a valid state, prevent submitting when its blank - araujobarret

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[DIA-198]: https://artsyproduct.atlassian.net/browse/DIA-198?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Onboarding/OnboardingLogin.tests.tsx [hunk 1]
```diff
@@ -38,94 +37,99 @@ describe("OnboardingLogin", () => {
...
+    expect(mockValidateForm).toHaveBeenCalledTimes(1)
+
+    fireEvent.changeText(screen.getByTestId("password"), "password 2")
+    fireEvent(screen.getByTestId("password"), "submitEditing")
+    waitFor(() => {
+      expect(mockValidateForm).toHaveBeenCalledTimes(2)
     })
   })
+
+  it("does not submit the form given a submission in an empty form state", () => {
```
Comment (araujobarret) [inline src/app/Scenes/Onboarding/OnboardingLogin.tests.tsx:113]: All other test cases are just refactoring/simplification(no need for many describes imo) but this one is the one that prevents the bug mentioned from being re-introduced.

File: src/app/Scenes/Onboarding/OnboardingLogin.tsx [hunk 2]
```diff
@@ -166,7 +175,7 @@ export const OnboardingLoginWithEmailForm: React.FC<OnboardingLoginProps> = ({
             onPress={handleSubmit}
             block
             haptic="impactMedium"
-            disabled={!(isValid && dirty) || isSubmitting} // isSubmitting to prevent weird appearances of the errors caused by async submiting
+            disabled={!isValid || !valuesNotEmpty}
```
Comment (araujobarret) [inline src/app/Scenes/Onboarding/OnboardingLogin.tsx:178]: `isSubmitting` wasn't necessary with the changes made, couldn't reproduce any weird UI behaviour with it

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (fix: disable the cta on the login screen until the form is in a valid state, prevent submitting when its blank - araujobarret)
- undefined (changes)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4ca25484961512fce123bd9995cbffc404fa660d
</p>

File: src/app/Scenes/Onboarding/OnboardingLogin.tsx [hunk 3]
```diff
@@ -115,6 +118,7 @@ export const OnboardingLoginWithEmailForm: React.FC<OnboardingLoginProps> = ({
               // enable autofill of login details from the device keychain.
               textContentType="username"
               error={errors.email}
+              testID="email-address"
```
Comment (egdbear) [inline src/app/Scenes/Onboarding/OnboardingLogin.tsx:121]: At the moment, I was wondering why this property is called testID, but then I realized that we are actually using it for testing purposes 😅

**Review by egdbear — APPROVED** [review]

File: src/app/Scenes/Onboarding/OnboardingLogin.tsx (hunk 3 above)
Comment (araujobarret) [reply src/app/Scenes/Onboarding/OnboardingLogin.tsx:121]: yep, tried selecting via accessibility props but none of them were exposed, bonus point is these ids can be used in the future for automated tests

**Pull request merged**